File: app/src/main/java/com/worldventures/dreamtrips/wallet/ui/settings/help/video/holder/VideoHolderFactoryImpl.java
package com.worldventures.dreamtrips.wallet.ui.settings.help.video.holder;

import android.view.ViewGroup;

import com.worldventures.dreamtrips.R;
import com.worldventures.dreamtrips.wallet.ui.common.adapter.BaseHolder;
import com.worldventures.dreamtrips.wallet.ui.settings.help.video.delegate.WalletVideoCallback;
import com.worldventures.dreamtrips.wallet.ui.settings.help.video.model.WalletVideoModel;

import static android.databinding.DataBindingUtil.bind;
import static android.view.LayoutInflater.from;

public class VideoHolderFactoryImpl implements VideoTypeFactory {

   private final WalletVideoCallback videoActionsCallback;
   private final WalletVideoHolderDelegate videoHolderDelegate;

   public VideoHolderFactoryImpl(WalletVideoCallback videoActionsCallback, WalletVideoHolderDelegate videoHolderDelegate) {
      this.videoHolderDelegate = videoHolderDelegate;
      this.videoActionsCallback = videoActionsCallback;
   }

   @Override
   public int type(WalletVideoModel video) {
      return R.layout.adapter_item_video;
   }

   @Override
   public BaseHolder holder(ViewGroup parent, int viewType) {
      if (viewType == R.layout.adapter_item_video) {
         return new WalletVideoHolder(bind(from(parent.getContext()).inflate(viewType, parent, false)),
               videoActionsCallback, videoHolderDelegate);
      } else {
         throw new IllegalArgumentException();
      }
   }
}

File: wallet/src/main/java/com/worldventures/wallet/ui/settings/help/video/holder/VideoHolderFactoryImpl.kt
package com.worldventures.wallet.ui.settings.help.video.holder

import android.databinding.DataBindingUtil.inflate
import android.view.LayoutInflater.from
import android.view.ViewGroup
import com.worldventures.wallet.R
import com.worldventures.wallet.ui.common.adapter.BaseHolder
import com.worldventures.wallet.ui.settings.help.video.delegate.WalletVideoCallback
import com.worldventures.wallet.ui.settings.help.video.model.WalletVideoModel

class VideoHolderFactoryImpl(private val videoActionsCallback: WalletVideoCallback,
                             private val videoHolderDelegate: WalletVideoHolderDelegate) : VideoTypeFactory {

   override fun type(video: WalletVideoModel) = R.layout.item_wallet_video

   override fun holder(parent: ViewGroup, viewType: Int): BaseHolder<*> {
      return when (viewType) {
         R.layout.item_wallet_video ->
            WalletVideoHolder(inflate(from(parent.context), viewType, parent, false), videoActionsCallback, videoHolderDelegate)
         else -> throw IllegalArgumentException()
      }
   }
}
